kvserver: add rac2 pull mode integration test variations

kvoli opened the pull request:
Note, pull mode still cannot be enabled without #132125, so in reality
these tests are operating in push mode, but either applying to elastic
or all replication traffic for now. e.g., the basic variations are now:

```
+----------+----------+------------------+-----------+
| Protocol | Encoding | Mode             | Push/Pull |
+----------+----------+------------------+-----------+
| v2       | v1       | apply_to_elastic | push      |
| v2       | v1       | apply_to_all     | push      |
| v2       | v2       | apply_to_elastic | push      |
| v2       | v2       | apply_to_all     | push      |
+----------+----------+------------------+-----------+
```



Once #132125 merges, these tests will additionally be exercising pull
mode, e.g., the basic variations will be:

```
+----------+----------+------------------+-----------+
| Protocol | Encoding | Mode             | Push/Pull |
+----------+----------+------------------+-----------+
| v2       | v1       | apply_to_elastic | push      |
| v2       | v1       | apply_to_all     | pull      |
| v2       | v2       | apply_to_elastic | push      |
| v2       | v2       | apply_to_all     | pull      |
+----------+----------+------------------+-----------+
```

See individual commit messages for more details on the changes
made in this PR. All relate to testing. The first commit is
mechanical.


Part of: #128040
Release note: None

kvoli committed:
kvserver: specify flow_control_integration_test_v2 encoding

Note to the reader, only files have moved without other changes.

We didn't specify the v2 encoding in the testdata file but did for v1
encoding.

Now specify both.

Part of: #128040
Release note: None

kvoli committed:
kvserver: add rac2 pull mode integration test variations

Introduce pull mode (send queue) variants to the existing
TestFlowControl.*V2 integration tests.

The tests specify `apply_to_all`, rather than `apply_to_elastic` for the
cluster setting `kvadmission.flow_control.mode` -- which enables pull
mode in raft and send queue management in RACv2.

These tests use their own file, as the behavior differs in some cases
between pull and push mode.

Part of: #128040
Release note: None

kvoli committed:
kvserver: allow waiting for + tokens available in TestFlowControl

We already allowed waiting for all tokens to be returned. Also allow
waiting for a specific number of tokens to be available, i.e., instead
of the limit, any arbitrary value.

Part of: #128040
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/132374)
<!-- Reviewable:end -->

sumeerbhola reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 14 of 14 files at r1, 58 of 58 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/132374#-:-O8vxOqiC2kcqKo84RhQ:bg7igkv)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @kvoli and @pav-kv)

___
*[`pkg/kv/kvserver/testdata/flow_control_integration_v2/admission_post_split_merge_v2_encoding_apply_to_elastic` line 1 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/132374#-O8vt0Cx8D3joyQQHTug:-O8vt0Cx8D3joyQQHTuh:bww66dh) ([raw file](https://github.com/cockroachdb/cockroach/blob/ade44aa2df0c18aa5d7deabc2799ccf4788124de/pkg/kv/kvserver/testdata/flow_control_integration_v2/admission_post_split_merge_v2_encoding_apply_to_elastic#L1)):*
> ```
> echo
> ```

nit: should this have a similar comment about all comments being reinterpreted as elastic, since presumably this test is explicitly sending elastic traffic only?


<!-- Sent from Reviewable.io -->

kvoli committed:
kvserver: skip TestFlowControlTransferLeaseV2 under duress

We are already going to skip this as part of #132125.

Unblock merging by skipping now as well.

Informs: #132125
Part of: #128040
Release note: None
